firmware/hw_layer/adc/adc_inputs.cpp

```diff
@@ -550,6 +582,19 @@ void initAdcInputs() {
 	adcStart(&ADC_FAST_DEVICE, NULL);
 	adcSTM32EnableTSVREFE(); // Internal temperature sensor
 
+	/* Enable this code only when you absolutly sure
+	 * that there is no possible errors from ADC */
+#if 0
+	/* All ADC use DMA and DMA calls end_cb from its IRQ
```

Comment:
imo an even better solution would be to trigger the fast adc with a timer, so that we collect perhaps 1 sample per degree, then compute the average exactly once at the end, instead of ~2 interrupts per sample

Reply:
but that's a problem for another PR

Reply:
There are few strange things with STM32 ADCs:
1. all ADCs share same IRQ vector.
2. while running continues conversion with DMA ADC still generates 2 irq for each batch conversion: on start and on end of conversion. I was not able to find way to disable this IRQ sources. Only disabling ADC vector at interrupt controller. But we still interested in error/overrun events.
3. sw logic rely on DMA end of transfer IRQ. We compute averages at DMA tranfer end callback. So these two ADC irq per batch conversion are totally useless and just wasting cpu time.